lms/djangoapps/program_enrollments/management/commands/send_program_course_nudge_email.py

```diff
@@ -0,0 +1,196 @@
+            'and [%s] distinct courses',
+            passed_grades.count(),
+            yesterday,
+            len(passing_user_ids),
+            len(passed_course_to_users_maps.keys()),
+        )
+
+        return passed_course_to_users_maps
+
+    def valid_course_run(self, course_run):
+        """
+        Check if a course run is in enrollable state.
+        """
+        return course_run['is_enrollable'] \
+            and course_run['is_marketable'] \
+            and course_run['marketing_url'] \
+            and course_run['status'] == COURSE_PUBLISHED
+
+    def get_course_run_to_suggest(self, programs_progress, completed_course_id):
+        """
+        Finds out enrollable course run from programs Generated by ProgramProgressMeter.
+
+        Returns: Suggested program and course_run dicts
+        """
+        for program in programs_progress:
+            for not_started_courses in program['not_started']:
+                for course_run in not_started_courses['course_runs']:
+                    if self.valid_course_run(course_run) and course_run['key'] != completed_course_id:
+                        return program, course_run
+        return None, None
+
+    def sort_programs(self, programs):
+        """
+        Sorts programs based on their revenue ranking.
+        """
+        sort_revenue_order = {
+            'MicroMasters': 1,
+            'Professional Program': 2,
+            'Professional Certificate': 3,
+            'XSeries': 4,
+            'Masters': 5,
+            'MicroBachelors': 6,
+        }
+        for program in programs:
+            program['sort_revenue_order'] = sort_revenue_order.get(program['type'], 7)
+
+        return sorted(programs, key=itemgetter('sort_revenue_order'))
+
+    def get_course_run(self, program, course_run_id):
+        """
+        get course run from a program.
+        """
+        for course in program['courses']:
+            for course_run in course['course_runs']:
+                if course_run['key'] == course_run_id:
+                    return course_run
+
+    def get_program(self, programs, program_progress):
+        """
+        get detailed program.
+        """
+        for program in programs:
+            if program['uuid'] == program_progress['uuid']:
+                return program
+
+    def emit_event(self, user, program, suggested_course_run, completed_course_run):
+        """
+         Emit the Segment event which will be used by Braze to send the email
+        """
+        event_properties = {
+            'COURSE_ONE_NAME': completed_course_run['title'],
+            'PROGRAM_TYPE': program['type'],
+            'PROGRAM_TITLE': program['title'],
+            'COURSE_TWO_NAME': suggested_course_run['title'],
+            'COURSE_TWO_SHORT_DESCRIPTION': suggested_course_run['short_description'],
+            'COURSE_TWO_LINK': urljoin(settings.MKTG_URLS.get('ROOT'), suggested_course_run['marketing_url']),
+            'USERNAME': user.username,
```

Comment:
username is not PII ... right?

Reply:
Username is PII info

Reply:
As it was extra property, I have removed it.